File: web2-project/src/app/model/consumer.ts
import { Address } from "./address";

export enum Type{
    Residential,
    Commercial
}

export class Consumer {
    public consumerID: number;
    public firstName: string;
    public lastName: string;
    public address: Address;
    public priority: number;
    public phoneNumber: number;
    public type: Type;

    constructor(firstName: string, lastName: string, address: Address, phoneNumber: number, type: Type){
        this.firstName = firstName;
        this.lastName = lastName;
        this.address = address;
        this.priority = address.priority;
        this.phoneNumber = phoneNumber;
        this.type = type;
    }
}

File: Web2BackEnd/Web2BackEnd/Models/Consumer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Web2BackEnd.Models
{
	public class Consumer : GenericModel
	{
		[Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int ConsumerID { get; set; }
		[Required]
		public string FirstName { get; set; }
		[Required]
		public string LastName { get; set; }
		[Required]
		public Address Address { get; set; }
		[Required]
		public int PhoneNumber { get; set; }
		[Required]
		public Type Type { get; set; }
		[Required]
		public int Priority { get; set; }
	}

	public enum Type{
		Residential,
		Commercial
	}
}
